addon/services/file-manager.js

```diff
@@ -1,165 +1,318 @@
+     * rename this file to avoid conflict. If replace, the existing file is
+     * destroyed.
+     * @return {Promise} Promise that resolves to the the new `file` model or
+     * rejects with an error message.
+     */
     copy(file, targetFolder, options={}) {
-        options.copy = true;
+        if (!options.data) {
+            options.data = {};
+        }
+        options.data.action = 'copy';
         return this.move(file, targetFolder, options);
     },
 
-    deleteFile(file) {
+    /**
+     * Delete a file or folder
+     *
+     * @method deleteFile
+     * @param {file} file `file` model to delete.
+     * @param {Object} [options] Options hash
+     * @param {Object} [options.query] Key-value hash of query parameters to
+     * add to the request URL.
+     * @param {Object} [options.data] Payload to be sent.
+     * @return {Promise} Promise that resolves on success or rejects with an
+     * error message.
+     */
+    deleteFile(file, options = {}) {
         let url = file.get('links').delete;
-        let p = this._waterbutlerRequest('DELETE', url);
-        return p.then(() => {
-            let parentFolder = file.get('parentFolder');
-            if (parentFolder) {
-                parentFolder.get('files').removeObject(file);
-            }
-        });
+        let p = this._waterbutlerRequest('DELETE', url, options);
+        return p.then(() => file.get('parentFolder').then((parent) => {
+                if (parent) {
+                    return this._reloadModel(parent.get('files'));
+                } else {
+                    this.get('store').unloadRecord(file);
+                    return true;
+                }
+            })
+        );
     },
 
-    ///////////////////// Privates /////////////////////
-
-    _waterbutlerRequest(method, url, queryParams=null, data=null) {
-        if (queryParams) {
-            let queryString = Ember.$.param(queryParams);
+    /**
+     * Make a Waterbutler request
+     *
+     * @method _waterbutlerRequest
+     * @private
+     * @param {String} method HTTP method for the request.
+     * @param {String} url Waterbutler URL.
+     * @param {Object} [options] Options hash
+     * @param {Object} [options.query] Key-value hash of query parameters to
+     * add to the request URL.
+     * @param {Object} [options.data] Payload to be sent.
+     * @return {Promise} Promise that resolves to the data returned from the
+     * server on success, or rejects with an error message.
+     */
+    _waterbutlerRequest(method, url, options = {}) {
+        if (options.query) {
+            let queryString = Ember.$.param(options.query);
             url = `${url}?${queryString}`;
         }
-        let sessionData = this.get('session').get('data').authenticated;
-        let accessToken = sessionData.attributes.accessToken;
+
+        let headers = {};
+        this.get('session').authorize('authorizer:osf-token', (headerName, content) => {
```

Comment:
This about tracks with what I saw of ember simple auth- not as supportive of cookies as we would like, but it may be what we're stuck with. Thanks.